state/txindex/kv/kv.go

```diff
@@ -46,7 +46,7 @@ func (txi *TxIndex) Get(hash []byte) (*abci.TxResult, error) {
 
 	rawBytes, err := txi.store.Get(hash)
 	if err != nil {
-		panic(err)
+		return nil, nil
```

Comment:
the caller of the function expects a nil value in case nothing is found (look at the function description):
```
// Get gets transaction from the TxIndex storage and returns it or nil if the
// transaction is not found.
```

Specifically the caller code:
```
if res == nil {
		return nil, fmt.Errorf("tx (%X) not found", hash)
	}
```

Reply:
OK. Same effect if we return an error (which makes more sense to me and I think is better practice)

Usually the caller checks for err, and otherwise assumes the res to be non-nil

Reply:
Agree. didn't write the calling code (as it was copied from tm) and didn't want to change current function assumption currently.